src/Pages/ProductListing/FiltersContainer.jsx

```diff
@@ -0,0 +1,252 @@
+    }, [response]);
+
+    useEffect(() => {
+        if(data?.categories?.length && ctgrySelected.ctgryId!==0){
+            let fltrArr = data.categories.filter(el=> el.id == ctgrySelected.ctgryId)
+            fltrArr[0].productTags.map((cVal,cIndx) =>{
+                setPrdctTags(prevState => [...prevState, { id:cIndx, prdctTagTile:cVal, isSelected:false} ] )
+            })  
+            fltrArr[0].brandfilters.map((cVal,cIndx) =>{
+                setBrands(prevState => [...prevState, { id:cIndx, brandName:cVal, isSelected:false} ] )
+            })       
+        }            
+    }, [ctgrySelected]);
+
+    useEffect(() => {
+        if (props.clear) {
+            setCtgrySelected({indx:-1, ctgryId:0})
+            setPrdctTags([]);
+            setBrands([])
+        }
+    }, [props]);
+    
+    const handleCategory = (ctgry, cIndx) =>{
+        dispatch({ type: "CATEGORY" ,payload:ctgry.title })
+        setCtgrySelected({indx:cIndx, ctgryId:Number(ctgry.id)})
+        setPrdctTags([]);
+    }
+
+    const handlebrandSearch = (e) =>{
+        setBrandSearchKey(e.target.value);
+        //FUNCTIONALITY yet to be implmented 
+        //Commented for future perspective
+        //return items.filter((item) => {
+            /*
+            // in here we check if our region is equal to our c state
+            // if it's equal to then only return the items that match
+            // if not return All the countries
+            */
+            //    if (item.region == filterParam) {
+            //        return searchParam.some((newItem) => {
+            //          return (
+            //            item[newItem]
+            //                .toString()
+            //                .toLowerCase()
+            //                .indexOf(q.toLowerCase()) > -1
+            //                     );
+            //                 });
+            //             } else if (filterParam == "All") {
+            //                 return searchParam.some((newItem) => {
+            //                     return (
+            //                         item[newItem]
+            //                             .toString()
+            //                             .toLowerCase()
+            //                             .indexOf(q.toLowerCase()) > -1
+            //                     );
+            //                 });
+            //             }
+            //         });
+    }
+
+    const handleProductsTagCheckBox = (prdctTag) =>{
+        let indx = prdctTags.findIndex(el=> el.prdctTagTile === prdctTag)
+        let prdctTagsArr = [...prdctTags];
+        let prdctTagTobeChanged = {...prdctTagsArr[indx]};
+        prdctTagTobeChanged.isSelected = !prdctTagTobeChanged.isSelected;
+        prdctTagsArr[indx] = prdctTagTobeChanged;
+        setPrdctTags([...prdctTagsArr])
+        const selectedProductTags = prdctTagsArr.filter(el=> el.isSelected === true).reduce((acc, cVal) => [...acc , cVal.prdctTagTile],[])
```

Comment:
Suggestion: Try to break this line in diff functions

Reply:
ok!